Source/Core/InputCommon/ControllerInterface/evdev/evdev.cpp

```diff
@@ -190,28 +301,21 @@ static void HotplugThreadFunc()
 
     if (strcmp(action, "remove") == 0)
     {
-      const auto it = s_devnode_name_map.find(devnode);
-      if (it == s_devnode_name_map.end())
-      {
-        // We don't know the name for this device, so it is probably not an evdev device.
-        continue;
-      }
+      std::shared_ptr<evdevDevice> ptr;
 
-      const std::string& name = it->second;
-      g_controller_interface.RemoveDevice([&name](const auto& device) {
-        return device->GetSource() == "evdev" && device->GetName() == name && !device->IsValid();
-      });
+      const auto it = s_devnode_objects.find(devnode);
```

Comment:
`devnode` is a `const char*`, while `s_devnode_objects`'s key is `std::string`. This will construct a temporary `std::string` , unless you make `s_devnode_objects` use a transparent comparator (`std::less<>`).

Reply:
This was also done before I touched the code. There's nothing too wrong with a temporary `std::string`.